cmd/genyaml/gen_kubectl_yaml.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v2"
+	"k8s.io/kubernetes/cmd/genutils"
+	"k8s.io/kubernetes/pkg/kubectl/cmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+type cmdOption struct {
+	Name         string
+	Shorthand    string `yaml:",omitempty"`
+	DefaultValue string `yaml:"default_value,omitempty"`
+	Usage        string `yaml:",omitempty"`
+}
+
+type cmdDoc struct {
+	Name             string
+	Synopsis         string       `yaml:",omitempty"`
+	Description      string       `yaml:",omitempty"`
+	Options          []cmdOption  `yaml:",omitempty"`
+	InheritedOptions []cmdOption	`yaml:"inherited_options,omitempty"`
+	Example          string       `yaml:",omitempty"`
+	SeeAlso          []string     `yaml:"see_also,omitempty"`
+}
+
+func main() {
```

Comment:
Yeah, getting a GenYAMLTree into cobra was the route that I was going to
take personally.

I'll think about whether we should just take this as-is in the meantime. :)

On Mon, Mar 28, 2016 at 4:51 PM, Thomas Liu <notifications@github.com>
wrote:

> In cmd/genyaml/gen_kubectl_yaml.go
> <https://github.com/kubernetes/kubernetes/pull/23526#discussion_r57653526>
> :
>
> > +	Shorthand    string `yaml:",omitempty"`
> > +	DefaultValue string `yaml:"default_value,omitempty"`
> > +	Usage        string `yaml:",omitempty"`
> > +}
> > +
> > +type cmdDoc struct {
> > +	Name             string
> > +	Synopsis         string       `yaml:",omitempty"`
> > +	Description      string       `yaml:",omitempty"`
> > +	Options          []cmdOption  `yaml:",omitempty"`
> > +	InheritedOptions []cmdOption	`yaml:"inherited_options,omitempty"`
> > +	Example          string       `yaml:",omitempty"`
> > +	SeeAlso          []string     `yaml:"see_also,omitempty"`
> > +}
> > +
> > +func main() {
>
> Thanks for the feedback! Cobra at the moment doesn't seem to have a
> convenient GenYamlTree function, so I based this off of this old gen man
> pages script for the time being:
>
>
> https://github.com/kubernetes/kubernetes/blob/643cb7a1c7499df4e569f4f0fbd3b18c0c4e63ce/cmd/genman/gen_kubectl_man.go
>
> As a side note, since the time of that genman script's writing, cobra now
> has built-in man page generation (GenManTree function, see:
> spf13/hugo#1369 <https://github.com/spf13/hugo/issues/1369>), so it might
> be possible to clean up that script and really easily extend man page
> generation to other kube commands (I can look into this if you'd like!).
>
> Regarding the current code, maybe another option would be to suggest a PR
> adding a GenYamlTree function to the cobra repo, and if it does get
> approved, use that here in place of the messy custom logic? Though that
> might take some time.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23526/files/0f08dc4bb7d601f9e2021bebbfa8878af7e201ab..c81f0a7b8222456d74098ffb8cd9cf09531e7a3f#r57653526>
>